### ISSUE TITLE
False positive for `MissingClosureParamType` on resource parameters

### ISSUE THREAD

**Issue by Radiergummi** [vimeo/psalm#4033]
**Environment**  
 - PHP `7.4.9`
 - Psalm `3.14.1@9822043ca46d6682b76097bfa97d7c450eef9e90`

**Summary**  
Psalm fails to recognize a closure parameter having the `resource` type, despite the PHPDoc comment annotation being present.
When refactoring the code into a named function, the issue doesn't occur (this is a non-workable solution for me though, and I guess this is a Psalm bug anyway).

**Steps to reproduce**  
The following example demonstrates this:
```php
/**
 * @param resource $handle
 */
$foo = function ($handle): void {};
```

On running Psalm with level 1, this yields:
```
ERROR: MissingClosureParamType - src/functions.php:12:18 - Parameter $handle has no provided type (see https://psalm.dev/153)
$foo = function ($handle): void {
```

I'd expect this to work, though, as using `resource` as a type hint is currently not supported in PHP.

**Comment by Radiergummi** [comment]
https://psalm.dev/r/1544e87bea

**Comment by Radiergummi** [comment]
@muglug thank you, that did it, allthough I don't find this too obvious. Wouldn't it make more sense to put the documentation comment on the property containing the closure?
